### ISSUE TITLE
/tests/console.html repl NameError when referencing global

### ISSUE THREAD

**Issue by perkinslr** [brython-dev/brython#622]
If you define 2 functions, 
 
    def a():
        return b()

    def b():
        return 5

And then try to call `a`, it fails with `NameError: name 'b' is not defined`.  If you define `b` first, it works as expected.

**Comment by kikocorreoso** [comment]
I've tested this and it is working. Could you provide a reproducible example?

If it is working, could we close this issue?

Thanks.

**Comment by PierreQuentel** [comment]
I also tested, and it fails for me with the same error message.

**Comment by kikocorreoso** [comment]
For me it works with the version on the site (brython.info) on the [console ](http://brython.info/tests/console.html?lang=en)and the [editor](http://brython.info/tests/editor.html?lang=en).

*version_info.js* says:

```
__BRYTHON__.implementation = [3, 3, 2, 'final', 0]
__BRYTHON__.__MAGIC__ = "3.3.2"
__BRYTHON__.version_info = [3, 3, 0, 'alpha', 0]
```

I think it is the latest stable release. ¿?

**Comment by PierreQuentel** [comment]
It works with the editor, but I just made the test on the console at brython.info : 

    Brython 3.3.2 on Netscape 5.0 (Windows)
    >>> def b():
    ...  return a()
    ... 
    >>> def a():
    ...  return 1
    ... 
    >>> b()
    Traceback (most recent call last):
      module __main__ line 145
        traceback.print_exc()
      module <module> line 1
        b()
      module <module> line 2
        return a()NameError: name 'a' is not defined
    >>>

**Comment by perkinslr** [comment]
@kikocorreoso You probably pasted the example code into the editor?  That makes it work.  Try pasting in `a` and executing it before defining `b`.  It looks like `func.globals` or whatever the brython equivalent is, is not being preserved in the repl.

**Comment by kikocorreoso** [comment]
@perkinslr very true :-(

**Comment by PierreQuentel** [comment]
The bug should be fixed by commit 86fbe1c4b4bdea1352be910795ff55a60168e760.

Can you try and close the issue if this is ok ?
